### ISSUE TITLE
Player.delete() doesn't unqueue audio

### ISSUE THREAD

**Issue by RiedleroD** [pyglet/pyglet#256]
When [Player.delete()](https://pyglet.readthedocs.io/en/latest/modules/media.html#pyglet.media.player.Player.delete) is called, it doesn't unqueue the queued songs, and therefore blocks any queued [StreamingSource](https://pyglet.readthedocs.io/en/latest/modules/media.html#pyglet.media.StreamingSource) from playing. [Player.next_source()](https://pyglet.readthedocs.io/en/latest/modules/media.html#pyglet.media.player.Player.next_source) has to be called too to unqueue the StreamingSource.

That's unintuitive behaviour since deleting the player would prevent it from playing again, no? It just blocks the StreamingSource.

**Comment by benmoran56** [comment]
Thanks for opening the ticket. If possible, could you show a small code example?

**Comment by RiedleroD** [comment]
```
import pyglet
from time import sleep
mus=pyglet.media.load("sample.ogg",streaming=True)
pl=mus.play()
sleep(0.5)
pl.delete()
sleep(0.5)
pl=mus.play()
sleep(0.5)
pl.delete()
mus.delete()
```
what should happen: music plays for 0.5 seconds, then stops for 0.5sec, then starts over for 0.5sec.
what happens: music plays for 0.5sec, then stops for 0.5sec, then crashes with `pyglet.media.exceptions.MediaException: This source is already queued on a player.` on line 7.

**Comment by benmoran56** [comment]
Thanks for the clarification.  I'll have a look at this.

**Comment by benmoran56** [comment]
Sorry that this got lost along the way, but it's now fixed in the 1.5-maintenance branch, and will be in the next point release.
